projects/js-packages/charts/src/components/pie-semi-circle-chart/pie-semi-circle-chart.tsx

```diff
@@ -24,66 +24,66 @@ interface PieSemiCircleChartProps extends BaseChartProps< DataPointPercentage[]
 	 */
 	thickness?: number;
 
+	/**
+	 * Direction of chart rendering
+	 * true for clockwise, false for counter-clockwise
+	 */
+	clockwise?: boolean;
+
 	/**
 	 * Label text to display above the chart
 	 */
 	label?: string;
+
 	/**
 	 * Note text to display below the label
 	 */
 	note?: string;
-	/**
-	 * Direction of chart rendering
-	 * true for clockwise, false for counter-clockwise
-	 */
-	clockwise?: boolean;
 }
 
 type ArcData = PieArcDatum< DataPointPercentage >;
 
+/**
+ * Validates the semi-circle pie chart data
+ * @param data - The data to validate
+ * @return Object containing validation result and error message
+ */
+const validateData = ( data: DataPointPercentage[] ) => {
+	if ( ! data.length ) {
+		return { isValid: false, message: 'No data available' };
+	}
+
+	// Check for negative values
+	const hasNegativeValues = data.some( item => item.percentage < 0 || item.value < 0 );
+	if ( hasNegativeValues ) {
+		return { isValid: false, message: 'Invalid data: Negative values are not allowed' };
+	}
+
+	// Validate total percentage is greater than 0
+	const totalPercentage = data.reduce( ( sum, item ) => sum + item.percentage, 0 );
+	if ( totalPercentage <= 0 ) {
+		return { isValid: false, message: 'Invalid percentage total: Must be greater than 0' };
+	}
+
+	return { isValid: true, message: '' };
+};
+
 const PieSemiCircleChart: FC< PieSemiCircleChartProps > = ( {
 	data,
-	width,
+	width = 400,
```

Comment:
It would make it easier to read if we don't change the order of the params if necessary; in this case, I have to go through every param to see whether anything is changed. We should probably avoid these changes unless it'd an improvement on readability etc.

Same as the changes on L27.